include/envoy/http/filter.h

```diff
@@ -119,6 +119,42 @@ enum class FilterMetadataStatus {
   Continue,
 };
 
+/**
+ * An interface for a per-connection object that is created by a filter and then managed by the
+ * HTTP connection manager. This pushes singleton handling into the HTTP connection manager.
+ */
+
+class MutableHttpConnection;
```

Comment:
I don't think forward declaration is needed here, though see below for my next comment.